pinot-controller/src/main/java/org/apache/pinot/controller/BaseControllerStarter.java

```diff
@@ -252,10 +253,13 @@ public void init(PinotConfiguration pinotConfiguration)
       // ControllerStarter::start()}
       _helixResourceManager = createHelixResourceManager();
       // This executor service is used to do async tasks from multiget util or table rebalancing.
+      _numThreadPool = _config.getControllerNumThreadPool();
       _executorService =
-          Executors.newCachedThreadPool(new ThreadFactoryBuilder().setNameFormat("async-task-thread-%d").build());
+          Executors.newFixedThreadPool(_numThreadPool, new ThreadFactoryBuilder()
+                  .setNameFormat("async-task-thread-%d").build());
       _tenantRebalanceExecutorService =
```

Comment:
Not introduced in this PR, but having 2 thread pool is hard to manage. I think we can just remove the `_tenantRebalanceExecutorService` and always use `_executorService`. @shounakmk219 wdyt?

Reply:
We can. 
The tenant rebalance job is a long running activity which allows parallelism and now that we have a fixed pool we don't want to block critical operations by occupying threads from the shared pool for such long durations. 
Maybe we should cap the degree of parallelism on tenant rebalance to certain % of `numThreadPool`?

Reply:
Good point! Should we make it a different configurable then? Basically keeping 2 thread pools, but allow them to be configured separately?
To a second thought, should be consider moving all rebalance tasks to a different thread pool? Rebalance is in general long running tasks which doesn't require much CPU

Reply:
Makes sense